lib/prompts.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+var _ = require("lodash");
+var async = require("async");
+var inquirer = require("inquirer");
+
+/**
+ * Prompt user for input, validate, and add derived fields to final data object.
+ *
+ * @param {Object}    init      Initialization configuration (`prompts`, `derived`)
+ * @param {Function}  callback  Calls back with `(err, data)`
+ * @returns {void}
+ */
+module.exports = function (init, callback) {
+  if (!init) { return callback(new Error("Invalid init object")); }
+
+  // Mutate `prompts` into appropriate format.
+  var prompts = init.prompts || [];
+  if (!_.isArray(prompts)) {
+    if (!_.isObject(prompts)) {
```

Comment:
Alternatively you could do this too. You don't get the early bailout but it is very explicit and doesn't have the nesting which IMO is a win.

```js
if (!init) { return callback(new Error("Invalid init object")); }

var prompts;
if (_.isArray(init.prompts)) {
  prompts = init.prompts;
} else if (_.isObject(init.prompts)) { 
  promps = _.map(prompts, function (val, key) {
    return _.extend({ name: key }, val);
  });
} else {
  return callback(new Error("Invalid prompts type: " + typeof prompts));
}
```